src/Blockcore.Indexer.Core/Storage/Mongo/MongoData.cs

```diff
@@ -79,48 +79,48 @@ public List<string> GetBlockIndexIndexes()
 
       public List<IndexView> GetIndexesBuildProgress()
       {
-            IMongoDatabase db = mongoDatabase.Client.GetDatabase("admin");
-            var command = new BsonDocument {
+         IMongoDatabase db = mongoDatabase.Client.GetDatabase("admin");
+         var command = new BsonDocument {
                { "currentOp", "1"},
             };
-            BsonDocument currentOp = db.RunCommand<BsonDocument>(command);
+         BsonDocument currentOp = db.RunCommand<BsonDocument>(command);
 
-            BsonElement inproc = currentOp.GetElement(0);
-            var arr = inproc.Value as BsonArray;
+         BsonElement inproc = currentOp.GetElement(0);
+         var arr = inproc.Value as BsonArray;
 
-            var ret = new List<IndexView>();
+         var ret = new List<IndexView>();
 
-            foreach (BsonValue bsonValue in arr)
+         foreach (BsonValue bsonValue in arr)
+         {
+            BsonElement? desc = bsonValue.AsBsonDocument?.GetElement("desc");
```

Comment:
I remember this was a bit of a specialized code handling to get it right, why change it? is this more generic and correct?

Reply:
From what I can see it's just null checks that I have added. Which lines of code are you referring to specifically?

Reply:
ok yeah that's fine its just whitespaces